flink/src/main/java/org/apache/iceberg/flink/IcebergTableSink.java

```diff
@@ -63,6 +63,7 @@ public SinkRuntimeProvider getSinkRuntimeProvider(Context context) {
         .tableLoader(tableLoader)
         .tableSchema(tableSchema)
         .equalityFieldColumns(equalityColumns)
+        .upsert(equalityColumns.size() > 0)
```

Comment:
I think we could enable `upsert` mode on flink sql when the primary key has been set

Reply:
The upsert mode will produce an extra DELETE for each INSERT operation,  that means the read scanning job will need to join more extra data which will effect the read performance.  If it's the CDC cases, all the events are produced as an UPDATE_BEFORE and an UPDATE_AFTER and the stream can gurantee the uniqueness when applying each events,  then we don't have to enable the upsert switch. 

I will suggest to introduce a table property named `write.upsert.enable`  for flink table, so that people could choose whether they need the real UPSERT.

Reply:
I ignore the CDC cases, make the upsert mode as configurable is makes sense